sunpy/map/mapbase.py

```diff
@@ -1598,24 +1598,22 @@ def save(self, filepath, filetype='auto', **kwargs):
         Parameters
         ----------
         filepath : `str`
-            Location to save file to.
+            Location to save the file to. The file extension determines the format.
         filetype : `str`, optional
-            Any supported file extension, defaults to ``"auto"``.
+            The file format to save the map in. Defaults to ``"auto"`` which infers
+            the format from the file extension. Supported formats include FITS, JP2, and ASDF.
         hdu_type : `~astropy.io.fits.hdu.base.ExtensionHDU` instance or class, optional
-            By default, a FITS file is written with the map in its primary HDU.
-            If a type is given, a new HDU of this type will be created.
-            If a HDU instance is given, its data and header will be updated from the map.
-            Then that HDU instance will be written to the file.
-            The example below uses `astropy.io.fits.CompImageHDU` to compress the map.
+            For FITS files, this specifies the type of HDU to write. By default, the map is saved
+            in the primary HDU. If an HDU type or instance is provided, the map data and header will
+            be written to that HDU. For example, `astropy.io.fits.CompImageHDU` can be used to compress the map.
         kwargs :
-            Any additional keyword arguments are passed to
-            `~sunpy.io._file_tools_write_file`.
+            Any additional keyword arguments are passed to `~sunpy.io._file_tools.write_file`.
```

Comment:
I think we should pass keywords to `AsdfFile.write_to` as well.
```suggestion
            Any additional keyword arguments are passed to `~sunpy.io._file_tools.write_file` or `asdf.AsdfFile.write_to`.
```